postgresql/postgresql_plugin.cc

```diff
@@ -0,0 +1,832 @@
+
+        std::vector<std::tuple<ColumnName, CellValue, Date> > rowValues;
+        if (ntuples > 0) {
+            std::vector<std::tuple<ColumnName, CellValue, Date> > cols;
+            for(int j = 0; j < nfields; j++) {
+                rowValues.emplace_back(ColumnName(PQfname(res, j)), getCellValueFromPostgres(res, 0, j), Date::Date::notADate());
+                POSTGRESQL_VERBOSE(printf("[%d,%d] %s %s\n", 0, j, PQgetvalue(res, 0, j), PQfname(res, j));)
+            }       
+        }
+
+        PQclear(res);
+        PQfinish(conn);
+
+        return ExpressionValue(rowValues);
+
+    }
+
+    /** Return wheter or not all columns names and info are known.
+    */
+    virtual bool hasColumnNames() const { return false; }
+
+};
+
+/*****************************************************************************/
+/* POSTGRESQL RECORDER DATASET                                                */
+/*****************************************************************************/
+
+struct PostgresqlRecorderDatasetConfig {
+    string databaseName;
+    int port;
+    string host;
+    string tableName;
+    bool createTable;
+
+    PostgresqlRecorderDatasetConfig() {
+        databaseName = "database";
+        port = 1234;
+        host = "localhost";
+        tableName = "mytable";
+        createTable = false;
+    }
+};
+
+DECLARE_STRUCTURE_DESCRIPTION(PostgresqlRecorderDatasetConfig);
+DEFINE_STRUCTURE_DESCRIPTION(PostgresqlRecorderDatasetConfig);
+
+PostgresqlRecorderDatasetConfigDescription::
+PostgresqlRecorderDatasetConfigDescription()
+{
+    addField("databaseName", &PostgresqlRecorderDatasetConfig::databaseName, "Name of the database to connect to.");
+    addField("port", &PostgresqlRecorderDatasetConfig::port, "Port of the database to connect to.", 1234);
+    addField("host", &PostgresqlRecorderDatasetConfig::host, "Address of the database to connect to ");
+    addField("tableName", &PostgresqlRecorderDatasetConfig::tableName, "Name of the table to be recorded into");
+    addField("createTable", &PostgresqlRecorderDatasetConfig::createTable, "Should we create the table when the dataset is created", true);
+}
+
+struct PostgresqlRecorderDataset: public Dataset {
+
+    PostgresqlRecorderDatasetConfig config_;
+    std::unordered_set<ColumnName> insertedColumns;
+
+    pg_conn* startConnection() 
+    {
+        return startPostgresqlConnection(config_.databaseName, config_.host, config_.port);
+    }
+
+    PostgresqlRecorderDataset(MldbServer * owner,
+                 PolyConfig config,
+                 const std::function<bool (const Json::Value &)> & onProgress)
+        : Dataset(owner)
+    {
+        config_ = config.params.convert<PostgresqlRecorderDatasetConfig>();
+
+        if (config_.createTable) {            
+
+            auto conn = startConnection();
+
+            //Drop table if exist
+            string dropDescription = "DROP TABLE IF EXISTS " + config_.tableName;
```

Comment:
Isn't this dangerous.  I would make it configurable in the config and set it to false by default.